File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
# import math
# def solution(progresses, speeds):
#     count = 1
#     answer = []
#     temp = []
#     leng = len(progresses)
#     for i in range(leng):
#         temp.append(math.ceil((100 - progresses[i]) / speeds[i]))
#     print(temp)
    
#     current = temp[0]
#     for j in range(1,len(temp)):
#         if current < temp[j]:
#             answer.append(count)
#             count = 1
#             current = temp[j]
#         else:
#             count = count + 1
#     answer.append(count)
#     return answer

import math

def solution(progresses, speeds):
    count = 1
    answer = []
    temp = []
    leng = len(progresses)
    for i in range(leng):
        temp.append(math.ceil((100 - progresses[i]) / speeds[i]))
    temp.reverse()
    print(temp)
    
    current = temp.pop()
    while temp:
        if current < temp[-1]:
            answer.append(count)
            count = 1
            current = temp.pop()
        else:
            count += 1
            temp.pop()
    answer.append(count)
    return answer

File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  let count = 1;
  const answer = [];
  const temp = [];
  const leng = progresses.length;
  
  for (let i = 0; i < leng; i++) {
    temp.push(Math.ceil((100 - progresses[i]) / speeds[i]));
  }
  
  let current = temp[0];
  
  for (let j = 1; j < temp.length; j++) {
    if (current < temp[j]) {
      answer.push(count);
      count = 1;
      current = temp[j];
    } else {
      count = count + 1;
    }
  }
  
  answer.push(count);
  return answer;
}

